### PR TITLE
Dev/sebie01

### PR DESCRIPTION
**Pull request by sebibi07**
Fixed typos in READ.ME and introduced dynamic input shapes in train_LSTM.py as well as minor bug fixes.

### CODE REVIEW

File: code/train_LSTM.py [hunk 1]
```diff
@@ -273,8 +265,9 @@ def LSTM_Model(X):
   print("Splitting dataset into training test...")
   X_train_entire, X_validation_entire, y_train_entire, y_validation_entire = train_test_split(all_examples, all_labels, test_size=FLAGS.test_size)
   print("Creating Recurrent LSTM model...")
-  lstm_model = LSTM_Model(X_train_entire)
+  lstm_model = LSTM_Model()
```
Comment (obitech) [inline code/train_LSTM.py:265]: Why is this changing?

File: code/train_LSTM.py [hunk 2]
```diff
@@ -232,28 +229,23 @@ def LSTM_Model(X):
...
   layer = Dropout(0.4)(layer)   
-  #bring back the reshape, lstm, and dropout
   layer = Reshape(( int(layer.shape[1]), int(layer.shape[2]) * int(layer.shape[3])))(layer)
   layer = LSTM(LSTM_COUNT, return_sequences=False)(layer)
   layer = Dropout(0.4)(layer)
 
-
-
-  # N_DENSE Dense Layers
+  #Dense Layers
```
Comment (obitech) [inline code/train_LSTM.py]: `s/#Dense Layers/# Dense Layers/`

File: code/train_LSTM.py [hunk 3]
```diff
@@ -203,24 +203,21 @@ def load_spectrogram(rootDir):
   return np.array(input_examples), np.array(input_labels)
 
 
-def LSTM_Model(X):
+def LSTM_Model():
   
-
-  input_shape = (X[0].shape[0], X[0].shape[1], X[0].shape[2])
+  #Setting input shape dynamically, taking values from vggish_params
```
Comment (obitech) [inline code/train_LSTM.py]: Make sure to add a space after `#`

File: code/train_LSTM.py [hunk 4]
```diff
@@ -37,7 +37,7 @@
 
 flags.DEFINE_integer('minibatch_size', 16, 'Number of Mini batches executed per epoch (batch).')
 
-flags.DEFINE_integer('num_classes', 7, 'Number of classes to train on')
+flags.DEFINE_integer('num_classes', 10, 'Number of classes to train on')
```
Comment (obitech) [inline code/train_LSTM.py]: Why increase this?

File: README.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ After every epoch a snapshot of the models weights and biases is saved on disk.
...
 
 ### New: Convolutional LSTM for Sequence Classification
-In train_LSTM.py we provide a Convolutional LSTM for audio event recognition. Similar to vggish_train.py it performs classification tasks on mel spectrograms. In contrast to vggish_train.py, it does not perform a classification for each spectrogram but analyzes array of matrices and then performs a classification on the entire sequence. C-LSTMs may outperform CNNs when data only contains sparse specific features that don't occure in every timestep.
+In train_LSTM.py we provide a Convolutional LSTM for audio event recognition. Similar to vggish_train.py it performs classification tasks on mel spectrograms. In contrast to vggish_train.py, it does not perform a classification for each spectrogram but analyzes array of matrices and then performs a single classification on the entire sequence. C-LSTMs may outperform CNNs when data only contains sparse specific features that don't occure in every timestep.
```
Comment (obitech) [inline README.md:43]: Couple of things:

1. Moving forward I would like to introduce a line break after 80 characters (you can set this in VSCode). It makes reviewing individual lines easier and is also more convenient for users with smaller resolutions.
2. If you reference files, please use code formatting:
> In `train_LSTM.py`we provide a ...
3. Typo:
>  but analyzes **an** array
4. Claims such as ` C-LSTMs may outperform CNNs when data only contains sparse specific features that don't occure in every timestep` could use a reference.

**Review by obitech — CHANGES_REQUESTED** [review]
Left a few comments!

Please use more descriptive commit messages, "Minor changes" is not acceptable. See: https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53
At this point you will have to [squash the commits](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-and-merge) into a single one, providing a meaningful message.

Also please choose an appropriate title for the PR.

File: code/train_LSTM.py (hunk 1 above)
Comment (sebibi07) [reply code/train_LSTM.py:265]: No need to pass as a param since we can necessary values from vggish_params.py

File: code/train_LSTM.py (hunk 4 above)
Comment (obitech) [reply code/train_LSTM.py]: Don't just resolve this without addressing my comment.

Comment (sebibi07) [reply code/train_LSTM.py]: Back to 7.

**Pull request merged**